src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2264,6 +2400,58 @@ def _ensure_container_insights_for_monitoring(cmd, addon):
                               validate=False, no_wait=False, subscription_id=subscription_id)
 
 
+def _ensure_aks_acr(cli_ctx,
+                    client_id,
+                    acr_name_or_id,
+                    subscription_id,
+                    detach=False):
+    from msrestazure.tools import is_valid_resource_id
+    # Check if the ACR exists by resource ID.
+    if is_valid_resource_id(acr_name_or_id):
```

Comment:
You should go through ACR management client. ACR SDK is multu-api client, with potential to work in other cloud env, while your hard coded `2019-05-01` might not be available outside of public cloud. Let us follow the right pattern.

Reply:
Good catch. Fixed in the new commit.